piccolo/apps/user/tables.py

```diff
@@ -50,9 +50,24 @@ def __init__(self, **kwargs):
         # Generating passwords upfront is expensive, so might need reworking.
         password = kwargs.get("password", None)
         if password:
+            if password.startswith("pbkdf2_sha256") and len(password) == 117:
+                logger.warning("Hashed password passed to the constructor")
+                raise ValueError("Do not pass hashed password")
+
             kwargs["password"] = self.__class__.hash_password(password)
+
         super().__init__(**kwargs)
 
+    @classmethod
+    def from_fixture(cls, data: t.Dict[str, t.Any]):
+        """Create a BaseUser instance from fixture"""
+        for kwargs in data["user"][cls.__name__]:
+            password = kwargs.get("password")
+            user = cls(**{k: v for k, v in kwargs.items() if k != "password"})
+            user.password = password
+
+            return user
```

Comment:
With the new format given by Daniel this method can be simpler. Something like this
```python
@classmethod
def from_fixture(cls, data: t.Dict[str, t.Any]) -> BaseUser:
    """Create a BaseUser instance from fixture"""
    password = data.get("password")
    user = cls(**{k: v for k, v in data.items() if k != "password"})
    user.password = password
    return user
```